### PR TITLE
[swift/en] Update examples to Swift 2.1

### PR DESCRIPTION
**Pull request by cbess**
- update to upgrade to Swift 2.1
- code cleanup

### CODE REVIEW

File: swift.html.markdown [hunk 1]
```diff
@@ -209,25 +209,25 @@ default: // required (in order to cover all possible input)
...
+:returns: A string containing the name and day value.
 */
 func greet(name: String, day: String) -> String {
     return "Hello \(name), today is \(day)."
 }
-greet("Bob", "Tuesday")
+greet("Bob", day: "Tuesday")
 
 // similar to above except for the function parameter behaviors
-func greet2(#requiredName: String, externalParamName localParamName: String) -> String {
```
Comment (vendethiel) [inline swift.html.markdown]: did they really remove that? :o

File: swift.html.markdown [hunk 2]
```diff
@@ -516,9 +526,9 @@ class MyShape: Rect {
         // Place a question mark after an optional property, method, or
         // subscript to gracefully ignore a nil value and return nil
         // instead of throwing a runtime error ("optional chaining").
-        if let allow = self.delegate?.canReshape?() {
+        if let reshape = self.delegate?.canReshape?() where reshape {
```
Comment (vendethiel) [inline swift.html.markdown]: please explain the where a bit more

Comment (cbess) [inline swift.html.markdown]: Done, added an explicit "where" example

File: swift.html.markdown (hunk 1 above)
Comment (cbess) [inline swift.html.markdown]: Indeed, iz gone. I'm glad they removed it, felt a little weird using "#" symbol. Might be confused with the "#" for build conditions.

**Comment by vendethiel** [comment]
Looks good :)! Thank you!

**Comment by vendethiel** [comment]
:cake:

**Pull request merged**